src/stormd/service.rs

```diff
@@ -197,11 +199,23 @@ impl Runtime {
 
                 let (container_id, instr) = match mesg {
                     // These should be processed by transfer service
-                    // TODO: Ensure that the incoming chunks references correct app id and message
-                    // id
-                    Messages::PushContainer(AppMsg { app: _, data }) => {
-                        (data.container_id(), CtlMsg::ProcessContainer(data))
-                    }
+                    Messages::PushContainer(AppMsg { app, data }) => match app {
+                        StormApp::RgbTransfers => (
+                            data.container_id(),
+                            CtlMsg::GetContainer(AddressedClientMsg {
+                                remote_id,
+                                client_id: None,
+                                data: AppContainer {
+                                    storm_app: StormApp::FileTransfer,
+                                    container_id: storm::ContainerFullId {
+                                        message_id: MesgId::default(),
+                                        container_id: data.container_id(),
+                                    },
+                                },
+                            }),
+                        ),
+                        _ => (data.container_id(), CtlMsg::ProcessContainer(data)),
+                    },
```

Comment:
We can't allow application-specific logic to happen within generic daemon code. There should be another way of fixing that in non-application-specific path.

I am currently sick and my head is not thinking well, so I can't do a suggestion right now - need to do more investigation on the invalid behavior later

Reply:
The change is related with this PR RGB-WG/rgb-node/pull/202. 

I handle the `consignment file` from the `rgb daemon`, I will try extract this logic and create a specific bus message to  **receive by rgb-storm integration**. Do you agree?

And.. I hope you feel better soon.